perfkitbenchmarker/cloudstack/cloudstack_disk.py

```diff
@@ -0,0 +1,128 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module containing classes related to CloudStack disks.
+
+Disks can be created, deleted, attached to VMs, and detached from VMs.
+Use 'gcloud compute disk-types list' to determine valid disk types.
+"""
+
+from perfkitbenchmarker import disk
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker import flags
+from perfkitbenchmarker.cloudstack import util
+
+import string
+
+FLAGS = flags.FLAGS
+
+
+class CloudStackDisk(disk.BaseDisk):
+  """Object representing a Cloudstack Disk."""
+
+
+  def __init__(self, disk_spec, name, zone_id, project_id=None):
+    super(CloudStackDisk, self).__init__(disk_spec)
```

Comment:
As noted in the defaults file, it doesn't look like this supports "disk_type"s. If this is the case, it should be documented and the user should probably get a warning if they pass in a disk type which isn't supported